unblob/report.py

```diff
@@ -205,6 +205,12 @@ def lowest(self):
         return min(self.percentages)
 
 
+@attr.define(kw_only=True, frozen=True)
+class EntropyReport(Report):
+    shannon: EntropyMeasurements
+    chi_square: EntropyMeasurements
```

Comment:
This makes sense (maybe using a different name), but it is a breaking change for `Report` users.
I mean, it is probably fine to break it, but we can also be backward compatible if an extra report is returned instead of wrapping it in the changed old one.

```py
class _RandomnessReport(Report): ...  # renamed from EntropyReport, now an implementation detail
class EntropyReport(_RandomnessReport): pass
class ChiSquareReport(_RandomnessReport): pass
```

Reply:
Pushed a fixup (93aec8d) where I use this approach and also rename all occurences of "entropy" to "randomness" to be accurate while keeping backward compatibility.

The `ExtractionConfig` is still initialized with `entropy_depth` so we don't break the API, but internally it's an alias to `randomness_depth`.

The CLI accepts both `--randomness-depth` and `--entropy-depth` for the same field, so we don't break CLI for existing users.

Idea is to keep backward compatibility while limiting confusion when devs work around `_RandomnessReport`s.

Reply:
I am really sorry to lead you into a rabbit hole.

I see, that it was not possible to just add the new chi-2 report, as `UnknowChunk` was already expected to contain an `EntropyReport`.

In the end, we are also back with wrapping the 2 reports into `RandomnessReport` with "measurements" inside (as it was originally, as can be seen in type annotation like `randomness: Optional[Tuple[EntropyReport, ChiSquareReport]]`), only without the original names.

I suggest using something like the original solution (one report with named measurements inside), or at least using a `RandomnessReport` class with `entropy` and `chi_square` fields in place of `Tuple[EntropyReport, ChiSquareReport]`.
As backward compatibility can not be maintained, so I would also make the further changes as well (`ExtractionConfig.randomness_depth`, and removing `--entropy-depth`).

Reply:
It's done. The commit message has been adjusted too.

Reply:
Cool! :)

----

As the commit message was mentioned, I looked at it, and I think this part could be updated with the *randomness* word in place of *entropy*:

> This entropy measure is introduced by modifying the EntropyReport class
> so that it contains two RandomnessMeasurements:
> - shannon: for Shannon entropy, which was already there
> - chi_square: for Chi Square entropy, which we introduce

Reply:
Fixed.